src/Common/src/System/Threading/Tasks/TaskValue.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Threading.Tasks
+{
+    /// <summary>Value type discrimated union for a TResult and a <see cref="Task{TResult}"/>.</summary>
+    /// <typeparam name="TResult">The type of the result.</typeparam>
+    internal struct TaskValue<TResult>
+    {
+        /// <summary>The task. this will be non-null iff the operation didn't complete successfully synchronously.</summary>
```

Comment:
Wikipedia seems happy with the lower-case. I'm not personally aware of a requirement to capitalize that. I also think the usage fits quite well.

http://en.wikipedia.org/wiki/If_and_only_if